framework/db/ExpressionInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace yii\db;
+
+/**
+ * Interface ExpressionInterface should be used to mark classes, that should be built
+ * in a special way.
+ *
+ * The database abstraction layer of Yii framework supports objects that implement this
+ * interface and will use [[ExpressionBuilderInterface]] to build them.
+ *
+ * The default implementation is a class [[Expression]].
+ *
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.14
+ */
+interface ExpressionInterface
```

Comment:
how about calling it QueryFragmentInterface and make Query implement in the future?

It would make us able to use subqueries everywhere (everywhere where we have expression support) and simplify the code. Otherwise we'll never support subqueries in non-obvious places like inside `order by`

Reply:
Not sure about naming, but in general it's a good idea originally suggested by @hiqsol [in previous PR](https://github.com/yiisoft/yii2/pull/14836#issuecomment-330166721)

Reply:
Will try to adopt it in next changes iteration